### ISSUE TITLE
SMS Subscription issue

### ISSUE THREAD

**Issue by wolfthefallen** [securestate/king-phisher#116]
## Issue Description
Not recieving sms messages when subscribed to campaign

## Reproduction Steps
1. Start King Phisher
2. Send subscribe to campaign
3. send test email
4. click link, and submit creds

## Environment Details
Host OS: Ubuntu

## Error Details / Stack Trace

```
2016-05-25 21:29:44,642 JobManager                                         ERROR    job debcbc98-7190-4e76-be5e-1b82fc35b5ff encountered an error and is not set to tolerate exceptions
2016-05-25 21:29:44,643 JobManager                                         INFO     deleting job with id: debcbc98-7190-4e76-be5e-1b82fc35b5ff and callback function: issue_alert
2016-05-25 21:29:57,306 AdvancedHTTPServer                                 INFO     x.x.x "POST /owa/auth/inbox.html HTTP/1.1" 200 -
2016-05-25 21:29:57,309 KingPhisher.Server.RequestHandler                  INFO     handling a page visit for campaign id: 251 from IP address: xxx.xxx.xxx.xxx
2016-05-25 21:29:57,328 JobManager                                         INFO     adding new job with id: 6d0addef-b36b-4916-a1be-2c4bfab00faa and callback function: issue_alert
2016-05-25 21:29:57,328 JobManager                                         DEBUG    executing job with id: 6d0addef-b36b-4916-a1be-2c4bfab00faa and callback function: issue_alert
2016-05-25 21:29:57,336 AdvancedHTTPServer                                 DEBUG    sending alert SMS message to (sing it 8675309) (Verizon)
2016-05-25 21:29:58,659 JobManager                                         ERROR    job 6d0addef-b36b-4916-a1be-2c4bfab00faa encountered an error and is not set to tolerate exceptions
2016-05-25 21:29:58,660 JobManager                                         INFO     deleting job with id: 6d0addef-b36b-4916-a1be-2c4bfab00faa and callback function: issue_alert

```

**Comment by zeroSteiner** [comment]
Hah thanks. Just to be certain, you did change the number to 8675309 to censor it, right? That wasn't the original value?

**Comment by guitarmanj** [comment]
Lolz


Sent from my T-Mobile 4G LTE Device


-------- Original message --------

**Comment by wolfthefallen** [comment]
yup to censor it :)

**Comment by zeroSteiner** [comment]
Alright I was able to reproduce this, looks like a problem with how Verizon has started to check the domain of the sender address.

```
Traceback (most recent call last):
  File "king_phisher/sms.py", line 151, in <module>
    main()
  File "king_phisher/sms.py", line 144, in main
    if send_sms(results.message, results.phone_number, results.carrier):
  File "king_phisher/sms.py", line 123, in send_sms
    smtp_connection.sendmail(from_address, [to_address], message.as_string())
  File "/usr/lib/python2.7/smtplib.py", line 731, in sendmail
    raise SMTPSenderRefused(code, resp, from_addr)
smtplib.SMTPSenderRefused: (553, '#5.1.8 Domain of sender address <donotreply@nowhere.com> does not exist', 'donotreply@nowhere.com')
```

**Comment by zeroSteiner** [comment]
Fixed in commit 0855981bed545513315fa1aea2de36087c8dcd20 which is now in the dev branch. The fix consisted of changing `donotreply@nowhere.com` to `sms@king-phisher.com`. While that email address is not real, Verizon seems to want a legitimate domain to be used for the sender.

**Comment by wolfthefallen** [comment]
@zeroSteiner Thank you for taking care of this one
